src/test/java/org/springframework/data/hazelcast/repository/query/QueryIT.java

```diff
@@ -831,4 +824,62 @@ public void findByLastnameListenableFuture()
         assertThat(result.get(), containsInAnyOrder(person1, person2));
     }
     
+    @Test
+    public void countDistinctByFirstname() {
+        //adding same Person twice to create duplicate
+        Person p = new Person();
+        p.setId("2020");
+        p.setFirstname("Sachin");
+        p.setLastname("Tendulkar");
+        this.personMap.put("1001", p);
+        this.personMap.put("1002", p);
+
+        final Long distinctPersons = this.personRepository.countDistinctByFirstname("Sachin");
```

Comment:
Would be good to have a test case proving that we indeed have a duplicate(here and in other cases, just for our sanity):

```suggestion
final Long persons = this.personRepository.countByFirstname("Sachin");
assertThat(persons, equalTo(2L));
final Long distinctPersons = this.personRepository.countDistinctByFirstname("Sachin");
```